ibis/expr/datatypes.py

```diff
@@ -458,6 +458,36 @@ def valid_literal(self, value):
         ) and lower <= value <= upper
 
 
+@parametric
+class Interval(SignedInteger):
+
+    __slots__ = 'unit',
+
+    _nbytes = 4
+
+    _valid_units = frozenset([
+        'Y',   # Year
+        'M',   # Month
+        'w',   # Week
+        'd',   # Day
+        'h',   # Hour
+        'm',   # Minute
+        's',   # Second
+        'ms',  # Millisecond
+        'us',  # Microsecond
+        'ns'   # Nanosecond
+    ])
+
+    def __init__(self, unit='s', nullable=True):
+        super(Interval, self).__init__(nullable=nullable)
+        if unit not in self._valid_units:
+            raise ValueError('Unsupported interval unit `{}`'.format(unit))
+        self.unit = unit
+
+    def valid_literal(self, value):
+        return isinstance(value, six.integer_types + (datetime.timedelta,))
```

Comment:
I think only `timedelta` instances are valid literals, otherwise that would mean we're giving a default unit to integers.